src/engine/enginebuffer.cpp

```diff
@@ -437,17 +440,11 @@ void EngineBuffer::requestSyncMode(SyncMode mode) {
 }
 
 void EngineBuffer::readToCrossfadeBuffer(const int iBufferSize) {
-    // During startup, scaler is null.
-    if (m_pScale == NULL) {
-        return;
-    }
     CSAMPLE* fadeout = m_pScale->getScaled(iBufferSize);
     SampleUtil::copy(m_pCrossfadeBuffer, fadeout, iBufferSize);
 
     // Restore the original position that was lost due to getScaled() above
     m_pReadAheadManager->notifySeek(m_filepos_play);
-    // Clear the current scale information
-    m_pScale->clear();
```

Comment:
In some places we clear the old scaler on some other places we clear the new fader. This leads to a double clears jut to be sure. 
I have unified that and moved to the lazy approach to clear only the new fader, just before it is used.  
Since readToCrossfadeBuffer() does not know if the scaler is used afterwards, the is no need to clear here.